base_layer/core/src/proof_of_work/proof_of_work.rs

```diff
@@ -89,6 +96,20 @@ impl Display for ProofOfWork {
     }
 }
 
+impl ConsensusEncoding for ProofOfWork {
+    fn consensus_encode<W: Write>(&self, writer: &mut W) -> Result<usize, io::Error> {
+        let mut written = writer.write_varint(self.pow_algo.as_u64())?;
```

Comment:
The `pow_algo` is an enum - could we not just treat it as a single byte instead of as a u64, similar to `impl ConsensusEncoding for OutputFeaturesVersion {`?
```suggestion
        writer.write_all(&[self.pow_algo as u8])?;
        let mut written = 1;
```